transform/app/tasks/utils/send.py

```diff
@@ -3,15 +3,66 @@
 import logging
 from typing import Optional
 
+import boto3
+from pyspark.sql import DataFrame
+
 from app.services.celery.task import CeleryTaskStatus
+from app.services.celery.task_statuses import FAILURE, SUCCESS
+from app.services.s3.send import send_spark_df as send_to_s3
+from app.services.solr.send import send_spark_df as send_to_solr
+from app.settings import settings
+from app.tasks.utils.s3_paths import extract_after_bucket
 from app.worker import celery
 
 logger = logging.getLogger(__name__)
 
 
 @celery.task(name="send_data")
-def send_data(prev_task_status: Optional[CeleryTaskStatus], req_body: dict) -> dict:
+def send_data(
+    prev_task_status: Optional[CeleryTaskStatus],
+    df_transformed: DataFrame,
+    s3_client: boto3.client,
+    req_body: dict,
+    file_path: str,
+    collection_name: str,
+) -> CeleryTaskStatus:
     """Task to send data to solr/s3"""
-    logger.info("Sending data...")
-    # TODO
-    return CeleryTaskStatus(status="toBeImplemented").dict()
+    logger.info("Starting data sending task...")
+
+    if not prev_task_status or prev_task_status.get("status") != SUCCESS:
+        error_reason = "Previous task failed or is missing."
+        logger.error(error_reason)
+        prev_task_status = CeleryTaskStatus(status=FAILURE, reason=error_reason).dict()
+        return prev_task_status
+
+    try:
+
+        instances = req_body.get("instances", [])
+        s3_instance = next(
+            (inst for inst in instances if inst.get("type") == "s3"), None
+        )
+        solr_instance = next(
+            (inst for inst in instances if inst.get("type") == "solr"), None
+        )
+        file_key = extract_after_bucket(file_path, req_body.get("dump_url", ""))
+
+        json_data = df_transformed.toJSON().collect()
+
+        if solr_instance:
+            solr_url = solr_instance.get("url")
+            solr_collections = settings.COLLECTIONS[collection_name]["SOLR_COL_NAMES"]
```

Comment:
I believe this task should have two use cases:
- for e.g./full /batch update -  send data to predefined S3/Solr that is specified in .env. 
- but for /dump endpoint there is `cols_prefix` for solr instances. We should use the prefix and send to those collections. Note that prefixes can be different per instance.